android/app/src/main/java/com/github/quarck/calnotify/monitorstorage/MonitorDatabase.kt

```diff
@@ -0,0 +1,189 @@
+            return Room.databaseBuilder(
+                context.applicationContext,
+                MonitorDatabase::class.java,
+                DATABASE_NAME
+            )
+                .openHelperFactory(CrSqliteRoomFactory())
+                .allowMainThreadQueries() // Match existing behavior - MonitorStorage uses sync queries
+                .addCallback(LoggingCallback())
+                .build()
+        }
+        
+        /**
+         * Migrate legacy SQLiteOpenHelper database to Room-compatible schema.
+         * 
+         * This runs BEFORE Room opens the database because Room validates schema
+         * before running any migrations. For pre-Room databases (no room_master_table),
+         * the schema must match exactly or Room will reject it.
+         * 
+         * Changes made:
+         * - Adds NOT NULL constraints to primary key columns (Room requirement)
+         * - Preserves the existing index
+         * - All data is preserved
+         */
+        private fun migrateLegacyDatabaseIfNeeded(context: Context) {
+            val dbFile = context.getDatabasePath(DATABASE_NAME)
+            if (!dbFile.exists()) {
+                DevLog.info(LOG_TAG, "No existing database - fresh install")
+                return
+            }
+            
+            val db = SQLiteDatabase.openDatabase(dbFile.absolutePath, null, SQLiteDatabase.OPEN_READWRITE)
+            try {
+                if (!isLegacyDatabase(db)) return
+                performLegacyMigration(db)
+            } finally {
+                db.close()
+            }
+        }
+        
+        private fun isLegacyDatabase(db: SQLiteDatabase): Boolean {
+            val hasTable = db.rawQuery(
+                "SELECT 1 FROM sqlite_master WHERE type='table' AND name='manualAlertsV1'", null
+            ).use { it.moveToFirst() }
+            
+            val hasRoomTable = db.rawQuery(
+                "SELECT 1 FROM sqlite_master WHERE type='table' AND name='room_master_table'", null
+            ).use { it.moveToFirst() }
+            
+            return when {
+                !hasTable -> { DevLog.info(LOG_TAG, "No legacy table - nothing to migrate"); false }
+                hasRoomTable -> { DevLog.info(LOG_TAG, "Already Room-managed"); false }
+                else -> { DevLog.info(LOG_TAG, "Legacy database detected"); true }
+            }
+        }
+        
+        private fun performLegacyMigration(db: SQLiteDatabase) {
+            val rowCountBefore = db.rawQuery("SELECT COUNT(*) FROM manualAlertsV1", null)
+                .use { if (it.moveToFirst()) it.getLong(0) else 0 }
+            DevLog.info(LOG_TAG, "Migrating $rowCountBefore rows to Room-compatible schema")
+            
+            db.beginTransaction()
+            try {
+                // Recreate table with NOT NULL on primary key columns
+                db.execSQL("""
+                    CREATE TABLE manualAlertsV1_new (
```

Comment:
can we get these column names from the companion object of the original to avoid typos?